### PR TITLE
Correctly classify keywords in heritage clauses

### LINKED ISSUE #89: Typescript highlighting after "extends" keyword
_From @mattacosta on February 16, 2016 0:16_

It seems everything after "extends" turns into an identifier, when it should just be the very next word since there's only single inheritance. This is a problem if you implement an interface...

vscode-implements-keyword

_Copied from original issue: Microsoft/vscode#3038_

### PR DESCRIPTION
**Pull request by anubmat**
Fixes #89

### CODE REVIEW

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+8 error node v4.3.1
+9 error npm  v2.14.12
+10 error path C:\Users\anubmat\TypeScript-TmLanguage\package.json
+11 error code ENOENT
+12 error errno -4058
+13 error syscall open
+14 error enoent ENOENT: no such file or directory, open 'C:\Users\anubmat\TypeScript-TmLanguage\package.json'
+14 error enoent This is most likely not a problem with npm itself
+14 error enoent and is related to npm not being able to find a file.
+15 verbose exit [ -4058, true ]
```
Comment (DanielRosenwasser) [inline npm-debug.log]: Remove this.

File: tests/cases/Issue89.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export class ^FileLogger ^extends /**/^One ^implements/**/ ^IDisposable{
+	^	
+}
+^
```
Comment (zhengbli) [inline tests/cases/Issue89.ts]: Add more tests for complex cases:
1. `extends` more than one things: `export class A extends B, C, D implements E, F, G {`
2. switch the positions of `extends` and `implements`

Comment (zhengbli) [inline tests/cases/Issue89.ts]: Also add a marker inside the `/**/` comment block to make sure they are parsed as comment

**Comment by zhengbli** [comment]
:+1:

**Pull request merged**